File: backend/app/schemas/ai_dashboard.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class AIDashboardRead(BaseModel):
    start: datetime
    end: datetime
    suggestions_total: int
    feedback_total: int
    accepted: int
    edited: int
    rejected: int
    adoption_rate: float
    by_suggestion_type: dict[str, dict[str, int]]
    workflow: dict[str, int]
    bailian_calls: int
    bailian_failures: int
    rag_queries_total: int
    rag_fallbacks: int
    rag_fallback_rate: float
    agent_feedback_total: int
    agent_feedback_by_action: dict[str, int]
    agent_reasoning_total: int
    agent_reasoning_failures: int
    agent_reasoning_failure_rate: float
    avg_task_duration_ms: int

File: frontend/src/types/aiDashboard.ts
export interface AIDashboard {
  start: string
  end: string
  suggestions_total: number
  feedback_total: number
  accepted: number
  edited: number
  rejected: number
  adoption_rate: number
  by_suggestion_type: Record<string, { suggestions: number; accepted: number; edited: number; rejected: number }>
  workflow: { succeeded: number; failed: number; waiting_human: number }
  bailian_calls: number
  bailian_failures: number
  rag_queries_total: number
  rag_fallbacks: number
  rag_fallback_rate: number
  agent_feedback_total: number
  agent_feedback_by_action: Record<string, number>
  agent_reasoning_total: number
  agent_reasoning_failures: number
  agent_reasoning_failure_rate: number
  avg_task_duration_ms: number
}
